quarkus-integration/quarkus-benchmark/deployment/src/main/java/ai/timefold/solver/benchmark/quarkus/deployment/TimefoldBenchmarkProcessor.java

```diff
@@ -40,16 +40,20 @@ HotDeploymentWatchedFileBuildItem watchSolverBenchmarkConfigXml() {
     }
 
     @BuildStep
-    @Record(ExecutionTime.STATIC_INIT)
-    void registerAdditionalBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans,
-            BuildProducer<SyntheticBeanBuildItem> syntheticBeans,
+    BenchmarkConfigBuildItem registerAdditionalBeans(BuildProducer<AdditionalBeanBuildItem> additionalBeans,
             BuildProducer<UnremovableBeanBuildItem> unremovableBeans,
-            SolverConfigBuildItem solverConfigBuildItem,
-            TimefoldBenchmarkRecorder recorder) {
-        if (solverConfigBuildItem.getSolverConfig() == null) {
+            SolverConfigBuildItem solverConfigBuildItem) {
+        // We don't support benchmarking for multiple solvers
+        if (solverConfigBuildItem.getAllSolverConfigurations().size() > 1) {
+            throw new ConfigurationException("When defining multiple solvers, the benchmark feature is not enabled. " +
+                    "Consider using separate <solverBenchmark> instances for evaluating different solver configurations. " +
+                    "You can find additional information by visiting the Timefold documentation page and navigating to the " +
+                    "\"Benchmarking and tweaking\" section.");
```

Comment:
```suggestion
            throw new ConfigurationException("When defining multiple solvers, the benchmark feature is not enabled. " +
                    "Consider using separate <solverBenchmark> instances for evaluating different solver configurations.");
```

I wouldn't go so far as to recommend specific sections of the documentation.
Documentation changes, sections move around, titles are renamed... these comments very quickly get obsolete.

Also, please use raw string (`"""`) and put each sentence on its own line. That way, you get to avoid the ugly string contatenation where it's not necessary.